app/lib/shared/search_service.dart

```diff
@@ -73,17 +73,69 @@ class PackageDocument extends Object with _$PackageDocumentSerializerMixin {
       _$PackageDocumentFromJson(json);
 }
 
+/// How search results should be ordered.
+enum SearchOrder {
+  /// Search score should be a weighted value of [text], [updated],
+  /// [popularity], [health] and [maintenance], ordered decreasing.
+  overall,
+
+  /// Search score should depend only on text match similarity, ordered
+  /// decreasing.
+  text,
+
+  /// Search order should be in decreasing last updated time.
+  updated,
+
+  /// Search order should be in decreasing popularity score.
+  popularity,
+
+  /// Search order should be in decreasing health score.
+  health,
+
+  /// Search order should be in decreasing maintenance score.
+  maintenance,
```

Comment:
What's the difference between health and maintenance? For me, not knowing what it means, I would assume it's the same as `health`.

Reply:
- health: How the code qualifies against our tests (e.g. strong mode, some linter rules, works with the tools, doesn't have too many warnings)? E.g. I can have a lot of code in it that is using a deprecated API and it will likely break -> bad health.

- maintenance: How frequently is the package updated with new versions? E.g. If I publish a new version every 2-3 months, that seems like a steady maintenance. If I publish a new version after 2 years of break, that should score lower, even though it may be fresh. (There is a fundamental difference here with `updated`, which only cares about the latest publish time.)

Reply:
Yeah but you can interpret update frequency in different ways: Frequent updates (might indicate active maintenance, or unstable package) or very rarely (might indicate stable package, or unmaintained package)!

Reply:
Good point. I'll remove it for now, it should be cheap to re-introduce it later.